// @progress/kendo-react-layout/drawer/Drawer.mjs
/**
 * @license
 *-------------------------------------------------------------------------------------------
 * Copyright © 2024 Progress Software Corporation. All rights reserved.
 * Licensed under commercial license. See LICENSE.md in the package root for more information
 *-------------------------------------------------------------------------------------------
 */
"use client";
import * as t from "react";
import e from "prop-types";
import { DrawerContext as S } from "./context/DrawerContext.mjs";
import { DrawerNavigation as I } from "./DrawerNavigation.mjs";
import { validatePackage as W, useDir as O, classNames as T } from "@progress/kendo-react-common";
import { packageMetadata as j } from "../package-metadata.mjs";
const l = t.forwardRef((d, u) => {
  W(j);
  const {
    expanded: n,
    mode: i,
    position: p,
    className: f,
    children: v,
    style: w,
    animation: y,
    mini: m,
    width: h,
    miniWidth: b,
    items: a,
    item: k,
    tabIndex: g,
    onOverlayClick: x,
    onSelect: o
  } = d, r = t.useRef(null), E = t.useCallback(
    () => {
      r.current && r.current.focus();
    },
    []
  );
  t.useImperativeHandle(u, () => ({
    element: r.current,
    focus: E
  }));
  const N = t.useCallback(
    (D, P, s) => {
      if (a && o) {
        const R = {
          itemTarget: D,
          itemIndex: P,
          syntheticEvent: s,
          nativeEvent: s && s.nativeEvent,
          target: void 0
        };
        o.call(void 0, R);
      }
    },
    [a, o]
  ), c = O(r, d.dir), C = T(
    {
      "k-drawer-container": !0,
      "k-drawer-expanded": n,
      "k-drawer-overlay": i === "overlay",
      "k-drawer-push": i === "push",
      "k-drawer-mini": m && !n
    },
    f
  );
  return /* @__PURE__ */ t.createElement(
    S.Provider,
    {
      value: {
        animation: y,
        expanded: n,
        mode: i,
        position: p,
        mini: m,
        dir: c,
        items: a,
        item: k,
        width: h,
        miniWidth: b,
        onOverlayClick: x,
        onSelect: N
      }
    },
    /* @__PURE__ */ t.createElement(
      "div",
      {
        className: C,
        ref: r,
        dir: c,
        style: w,
        tabIndex: g
      },
      a && /* @__PURE__ */ t.createElement(I, null),
      v
    )
  );
});
l.propTypes = {
  animation: e.any,
  expanded: e.bool,
  children: e.any,
  className: e.string,
  dir: e.string,
  mode: e.string,
  position: e.string,
  mini: e.bool,
  style: e.object,
  tabIndex: e.number,
  width: e.number,
  miniWidth: e.number,
  selected: e.number,
  onSelect: e.func,
  onOverlayClick: e.func
};
const H = {
  animation: !0,
  expanded: !1,
  mode: "overlay",
  position: "start",
  mini: !1,
  dir: "ltr",
  width: 240,
  miniWidth: 48
};
l.defaultProps = H;
l.displayName = "KendoDrawer";
export {
  l as Drawer
};
