// @progress/kendo-react-layout/stepper/Step.mjs
/**
 * @license
 *-------------------------------------------------------------------------------------------
 * Copyright © 2024 Progress Software Corporation. All rights reserved.
 * Licensed under commercial license. See LICENSE.md in the package root for more information
 *-------------------------------------------------------------------------------------------
 */
"use client";
import * as e from "react";
import a from "prop-types";
import { StepperContext as Z } from "./context/StepperContext.mjs";
import { focusFirstFocusableChild as ee, dispatchEvent as L, classNames as te, IconWrap as O, toIconName as ae } from "@progress/kendo-react-common";
import { useLocalization as se } from "@progress/kendo-react-intl";
import { checkOutlineIcon as ne, exclamationCircleIcon as oe } from "@progress/kendo-svg-icons";
import { DEFAULT_ANIMATION_DURATION as ce, NO_ANIMATION as le } from "./contants.mjs";
import { messages as ie, optionalText as re } from "./messages/index.mjs";
const E = e.forwardRef((R, V) => {
  const {
    // content
    children: z,
    className: h,
    current: d,
    disabled: n,
    focused: u,
    icon: r,
    svgIcon: C,
    index: s,
    isValid: t,
    label: o,
    optional: m,
    style: y,
    tabIndex: S,
    text: F,
    ...D
  } = R, {
    animationDuration: k,
    isVertical: b,
    item: H,
    linear: P,
    mode: A,
    numOfSteps: c,
    value: l,
    onChange: f,
    onFocus: I,
    successIcon: _,
    errorIcon: w,
    successSVGIcon: B,
    errorSVGIcon: G
  } = e.useContext(Z), p = e.useRef(null), M = e.useCallback(
    () => {
      p.current && ee(p.current);
    },
    []
  ), v = e.useCallback(
    () => ({
      element: p.current,
      focus: M
    }),
    [M]
  );
  e.useImperativeHandle(V, v);
  const N = !P || s === l - 1 || s === l || s === l + 1, x = A === "labels" || !!r && !!o, U = se(), W = ((i) => U.toLanguageString(i, ie[i]))(re), $ = typeof k == "number" ? k : k !== !1 ? ce : le, j = e.useCallback(
    (i) => {
      f && !n && L(
        f,
        i,
        v(),
        { value: s }
      );
    },
    [f, l, n]
  ), K = e.useCallback(
    (i) => {
      I && !n && L(
        I,
        i,
        v(),
        void 0
      );
    },
    [I, n]
  ), q = e.useMemo(
    () => te(
      "k-step",
      {
        "k-step-first": s === 0,
        "k-step-last": c && s === c - 1,
        "k-step-done": s < l,
        "k-step-current": d,
        "k-step-optional": m,
        "k-step-error": t !== void 0 && !t,
        "k-step-success": t
      },
      {
        "k-disabled": n,
        "k-focus": u
      },
      h
    ),
    [s, c, l, d, m, n, u, t, h]
  ), J = e.useMemo(
    () => ({
      maxWidth: b ? void 0 : `calc(100% / ${c})`,
      maxHeight: b ? `calc(100% / ${c})` : void 0,
      pointerEvents: N ? void 0 : "none",
      ...y
    }),
    [b, c, y, N]
  ), T = t ? _ : w, g = T ? /* @__PURE__ */ e.createElement(
    "span",
    {
      className: "k-step-indicator-icon " + T,
      "aria-hidden": "true"
    }
  ) : /* @__PURE__ */ e.createElement(
    O,
    {
      className: "k-step-indicator-icon",
      name: t ? "check-circle" : "exclamation-circle",
      icon: t ? B || ne : G || oe
    }
  ), Q = /* @__PURE__ */ e.createElement(e.Fragment, null, A !== "labels" ? /* @__PURE__ */ e.createElement(
    "span",
    {
      className: "k-step-indicator",
      "aria-hidden": !0,
      style: { transitionDuration: $ + "ms" }
    },
    r || C ? !x && t !== void 0 ? g : /* @__PURE__ */ e.createElement(O, { className: "k-step-indicator-icon", name: r && ae(r), icon: C }) : t !== void 0 ? g : /* @__PURE__ */ e.createElement("span", { className: "k-step-indicator-text" }, F || s + 1)
  ) : null), X = (o || x || m) && /* @__PURE__ */ e.createElement("span", { className: "k-step-label" }, o && /* @__PURE__ */ e.createElement("span", { className: "k-step-text" }, o), x && t !== void 0 && g, m && /* @__PURE__ */ e.createElement("span", { className: "k-step-label-optional" }, W)), Y = /* @__PURE__ */ e.createElement(e.Fragment, null, Q, X);
  return /* @__PURE__ */ e.createElement(
    "li",
    {
      ref: p,
      className: q,
      style: J,
      ...D
    },
    /* @__PURE__ */ e.createElement(
      "a",
      {
        className: "k-step-link",
        title: o || void 0,
        onClick: j,
        onFocus: K,
        tabIndex: S || (u ? 0 : -1),
        "aria-current": d,
        "aria-disabled": n || !N || void 0,
        "aria-invalid": t !== void 0 && !t || void 0
      },
      H ? z : Y
    )
  );
});
E.propTypes = {
  children: a.any,
  className: a.string,
  // content: PropTypes.any,
  current: a.bool,
  disabled: a.bool,
  icon: a.string,
  index: a.number,
  isValid: a.bool,
  label: a.string,
  optional: a.bool,
  style: a.object,
  tabIndex: a.number,
  text: a.string
};
const me = {
  tabIndex: 0
};
E.defaultProps = me;
E.displayName = "KendoStep";
export {
  E as Step
};
